File: OPVOnline/opv_2d/static/opv_2d/js/model/cloudiness-cell.ts
/*
* В данном файле содержится реализация класса, который соответствует клетке с информацией об облачности.
*/

class CloudinessCell {
	private geomagneticLatitude: number

	private mlt: number

	public static firstCloudinessCell = new CloudinessCell(0, 0)

	public static secondCloudinessCell = new CloudinessCell(0, 0)

	public static thirdCloudinessCell = new CloudinessCell(0, 0)

	public static fourthCloudinessCell = new CloudinessCell(0, 0)

	private constructor(geomagneticLatitude: number, mlt: number) {
		this.geomagneticLatitude = geomagneticLatitude
		this.mlt = mlt
	}

	public set(geomagneticLatitude: number, mlt: number): void {
		this.geomagneticLatitude = geomagneticLatitude
		this.mlt = mlt
	}

	public getStandardPolarDistance(displayWidth: number): number {
		return (90 - this.geomagneticLatitude) / 80 * displayWidth
	}

	public getStandardPolarAngle(): number {
		return (this.mlt - 6) * Math.PI / 12
	}

	public getStandardX(displayWidth: number): number {
		return this.getStandardPolarDistance(displayWidth)
			* Math.cos(this.getStandardPolarAngle())
	}

	public getStandardY(displayWidth: number): number {
		return this.getStandardPolarDistance(displayWidth)
			* Math.sin(this.getStandardPolarAngle())
	}

	public static getColor(value: number) {
		return new Color(220, 220, 220, value / 100)
	}
}

File: OPVSupportManager/CloudinessDrawer/cloudiness_cell.py
# В данном файле содержится класс, который соответствует фрагменту изображения облачности.

import math


class CloudinessCell:
    def __init__(self, geomagnetic_latitude, mlt):
        self.geomagnetic_latitude = geomagnetic_latitude
        self.mlt = mlt

    def set(self, geomagnetic_latitude, mlt):
        self.geomagnetic_latitude = geomagnetic_latitude
        self.mlt = mlt

    def get_standard_polar_distance(self, display_width):
        return (90 - self.geomagnetic_latitude) / 80 * display_width

    def get_standard_polar_angle(self):
        return (self.mlt - 6) * math.pi / 12

    def get_standard_x(self, display_width):
        return self.get_standard_polar_distance(display_width) * math.cos(self.get_standard_polar_angle())

    def get_standard_y(self, display_width):
        return self.get_standard_polar_distance(display_width) * math.sin(self.get_standard_polar_angle())
